checker/src/main/java/org/checkerframework/checker/signature/SignatureAnnotatedTypeFactory.java

```diff
@@ -106,47 +124,65 @@ public Void visitCompoundAssignment(CompoundAssignmentTree node, AnnotatedTypeMi
+                    ExpressionTree arg0 = tree.getArguments().get(0);
+                    ExpressionTree arg1 = tree.getArguments().get(1);
+                    if (arg0.getKind() == Tree.Kind.CHAR_LITERAL
+                            && arg1.getKind() == Tree.Kind.CHAR_LITERAL) {
+                        c1 = (char) ((LiteralTree) arg0).getValue();
+                        c2 = (char) ((LiteralTree) arg1).getValue();
+                    }
+                } else {
+                    ExpressionTree arg0 = tree.getArguments().get(0);
+                    ExpressionTree arg1 = tree.getArguments().get(1);
+                    if (arg0.getKind() == Tree.Kind.STRING_LITERAL
+                            && arg1.getKind() == Tree.Kind.STRING_LITERAL) {
+                        String const0 = (String) ((LiteralTree) arg0).getValue();
+                        String const1 = (String) ((LiteralTree) arg1).getValue();
+                        if (const0.length() == 1 && const1.length() == 1) {
+                            c1 = const0.charAt(0);
+                            c2 = const1.charAt(0);
+                        }
                     }
                 }
-            } else if (TreeUtils.isMethodInvocation(
-                    tree, replaceCharSequenceCharSequence, processingEnv)) {
-                ExpressionTree arg0 = tree.getArguments().get(0);
-                ExpressionTree arg1 = tree.getArguments().get(1);
-                if (arg0.getKind() == Tree.Kind.STRING_LITERAL
-                        && arg1.getKind() == Tree.Kind.STRING_LITERAL) {
-                    String const0 = (String) ((LiteralTree) arg0).getValue();
-                    String const1 = (String) ((LiteralTree) arg1).getValue();
+                if (!(c1 == ' ' || c2 == ' ')) {
                     ExpressionTree receiver = TreeUtils.getReceiverTree(tree);
                     final AnnotatedTypeMirror receiverType = getAnnotatedType(receiver);
-                    if (const0.equals(".")
-                            && const1.equals("/")
-                            && receiverType.getAnnotation(BinaryName.class) != null) {
-                        type.replaceAnnotation(INTERNAL_FORM);
-                    } else if (const0.equals("/")
-                            && const1.equals(".")
-                            && receiverType.getAnnotation(InternalForm.class) != null) {
-                        type.replaceAnnotation(BINARY_NAME);
+                    if (receiverType.getAnnotation(BinaryName.class) != null) {
+                        handleReplaceOnBinaryName(c1, c2, type);
+                    } else if (receiverType.getAnnotation(InternalForm.class) != null) {
+                        handleReplaceOnInternalForm(c1, c2, type);
                     }
                 }
             }
             return super.visitMethodInvocation(tree, type);
         }
+
+        /**
+         * Handles String.replace calls when the receiver is a {@link
```

Comment:
Please give a more specific description than "handles", which doesn't say what this does.

Once we have a nice library for doing conversions among string formats, we could get rid of all this code and direct users to use the library instead, which will be clearer and less error-prone.